### PR TITLE
Restore gradient value when Variable is unPickled

### PR DESCRIPTION
**Pull request by hitottiez**
We want to use Chainer in multi-process environment. We have tried to use multiprocessing.Queue to share a model object (```chainer.Chain```) between processes. However, this strategy didn't work because gradient value was not restored when ```Variable``` instance was unPickled.

This pull request wants to fix the problem by keeping gradient values after unPickling ```Variable```.

Please see below for the details.
https://groups.google.com/forum/#!topic/chainer-jp/KneGQWbaPtw

### CODE REVIEW

**Comment by t-abe** [comment]
Thank you for the PR.
Could you make a test to check pickled & unpickled Variable has .grad correctly?

**Comment by hitottiez** [comment]
We added tests to check pickled & unpickled Variable has .grad correctly.
Please see our modified code.

**Comment by t-abe** [comment]
Thank you.  LGTM :smile:

File: chainer/variable.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ class Variable(object):
             :class:`~chainer.Flag` for the detail of ternary flags.
 
     """
-    def __init__(self, data, volatile=flag.OFF, name=None):
+    def __init__(self, data, volatile=flag.OFF, name=None, grad=None):
```
Comment (beam2d) [inline chainer/variable.py]: Would you add a document for the new argument?

Comment (hitottiez) [inline chainer/variable.py]: We added a docstring for the new argument.
Please check it.

**Comment by beam2d** [comment]
LGTM

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**